### PR TITLE
Adding a useful topic of how a custom Import Entity can be defined.

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) adds a new topic on how a custom **Import Entity** can be declared and used for importing the data from CSV files.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.2/ext-best-practices/tutorials/custom-import-entity.html
- https://devdocs.magento.com/guides/v2.3/ext-best-practices/tutorials/custom-import-entity.html

## Links to Magento source code

Existing Import Entities

- Product Import Entity https://github.com/magento/magento2/blob/2.3/app/code/Magento/CatalogImportExport/Model/Import/Product.php
- Product Options https://github.com/magento/magento2/blob/2.3/app/code/Magento/CatalogImportExport/Model/Import/Product/Option.php

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+This tutorial shows you how to add a new import entity, that can be used to import data into your custom module's table, by extending the [Magento/ImportExport/Model/Import/Entity/AbstractEntity][0]{:target="_blank"} class. 
+The current import entities can be found in **System** > **Import**:
+
+- Advanced Pricing
+- Products
+- Customers and Addresses (single file)
+- Customers Main File
+- Customer Addresses
+
+For the beginning let's suppose that we have a custom table with the following structure:
```
Comment (erikmarr) [inline guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md]:
```suggestion
To begin, let's suppose we have a custom table with the following structure:
```

File: guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+
+| entity_id | name | duration |
+| --- | --- | --- |
+|           |  |  |
+
+## Step 1: Adding a new Entity Type
```
Comment (erikmarr) [inline guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md]:
```suggestion
## Step 1: Adding a New Entity Type
```

File: guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+
+```xml
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_ImportExport:etc/import.xsd">
+    <entity name="learning" label="Learning Courses Import" model="OrangeCompany\Learning\Model\Import\Courses"
+            behaviorModel="Magento\ImportExport\Model\Source\Import\Behavior\Basic" />
+</config>
+```
+
+As we extend the **Magento_ImportExport** module, we should also add a dependency to it in `module.xml` file.
```
Comment (erikmarr) [inline guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md]:
```suggestion
As we extend the **Magento_ImportExport** module, we should also add a dependency to it in the `module.xml` file.
```

File: guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+
+{% endcollapsible %}
+
+## Result
+
+As result, we should be able to see the new Entity Type:
+ 
+![Import Entity]({{ site.baseurl }}/common/images/ext-best-practices/import-entity.png)
+ 
+ Here is a sample csv file that can be used for importing data into our table.
```
Comment (erikmarr) [inline guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md]:
```suggestion
 Here is a sample csv file we can use to import data into our table.
```

File: guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+
+```text
+entity_id,name,duration
+,"First Course",90
+,"Second Course",120
+```
+
+{:.bs-callout .bs-callout-info}
+For updating the table's data, the `entity_id` value is required to be provided for the respectively row.
```
Comment (erikmarr) [inline guides/v2.2/ext-best-practices/tutorials/custom-import-entity.md]:
```suggestion
For updating the table's data, you must provide the `entity_id` value for each row.
```

**Comment by eduard13** [comment]
@erikmarr, thank you for reviewing this PR, the requested changes were applied.
@osrecio, please let me know if anything is needed from my side.
Thanks.

**Comment by erikmarr** [comment]
running tests

**Comment by erikmarr** [comment]
Thanks @eduard13!

**Pull request merged**